.github/workflows/fevm-contract-tests.yaml

```diff
@@ -0,0 +1,150 @@
+    steps:
+      - name: Checkout Test Repository
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+          repository: 'consensus-shipyard/fevm-contract-tests'
+          path: 'fevm-contract-tests'
+      - name: Checkout Fendermint
+        uses: actions/checkout@v2
+        with:
+          repository: 'consensus-shipyard/ipc'
+          submodules: 'recursive'
+          path: 'ipc'
+      - name: Install Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18.19.0
+      - name: Install dependencies
+        run: npm install
+        working-directory: fevm-contract-tests
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+      - name: Set up Rust cargo cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+      - name: Install cargo-make
+        run: |
+          if ! command -v cargo-make &> /dev/null
+          then
+              cargo install -f cargo-make
+          fi
+
+      - uses: actions/cache/save@v3
+        if: always()
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Install Foundry
+        working-directory: ipc/contracts
+        run: |
+          curl -L https://foundry.paradigm.xyz | bash
+          export PATH="$PATH:/home/runner/.config/.foundry/bin"
+          foundryup
+      - name: Create a docker build
+        id: docker-build
+        working-directory: ipc/fendermint
+        run: |
+          export PATH="$PATH:/home/runner/.config/.foundry/bin"
+          make docker-build
+      - name: Run a testnode
+        id: testnode
+        working-directory: ipc/
+        run: |
+          export BALANCE=10000000000
+          { out=$(FM_PULL_SKIP=true cargo make --makefile ./infra/fendermint/Makefile.toml testnode | tee /dev/fd/3); } 3>&1
+          private_key=$(echo $out | sed -e 's/\(.*\)\([a-f0-9]\{64\}\)/0x\2/' | grep 0x | head -c66)
+          echo "ROOT_PRIVATE_KEY=$private_key" >> "$GITHUB_OUTPUT"
+          cat $GITHUB_OUTPUT
+      - name: 'Create and fund accounts'
+        id: accounts
+        env:
+          NUM_ACCOUNTS: 5
+          ROOT_PRIVATE_KEY: ${{ steps.testnode.outputs.ROOT_PRIVATE_KEY }}
+        working-directory: fevm-contract-tests
+        run: |
+          npx hardhat --network local create-fund-accounts
+      - name: 'Run tests: web3.js SimpleCoin'
+        if: always()
```

Comment:
Why are these `if: always()` conditions needed?

Reply:
This is a preference that @raulk expressed to me but I think his idea is that we want to ensure that all the tests run to ensure we have as much informat as possible in the event we need to debug a problem